cmd/login.go

```diff
@@ -42,7 +42,7 @@ var loginCmd = &cobra.Command{
 		azureClientId, _ := cmd.Flags().GetString(flagnames.AzureClientId)
 		azureClientSecret, _ := cmd.Flags().GetString(flagnames.AzureClientSecret)
 
-		if azureClientId != "" && azureClientSecret == "" && federatedTokenFile == "" {
+		if !useGithubCredentials && azureClientId != "" && azureClientSecret == "" && federatedTokenFile == "" {
```

Comment:
```suggestion
		if azureClientId != "" && !useGithubCredentials && azureClientSecret == "" && federatedTokenFile == "" {
```
I found it easier to understand the `if` clause when `azureClientId` is the first conditions